### PR TITLE
feat: SMS/Whatsapp to attendee for teams

### LINKED ISSUE #13897: [CAL-3430] [CAL-3233] Bring back SMS reminders for teams
[https://threads.com/thread/34562092433/34562108801?s=4w4qJ3A4yEg6BpFDxe6syY](https://threads.com/thread/34562092433/34562108801?s=4w4qJ3A4yEg6BpFDxe6syY)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3233](https://linear.app/calcom/issue/CAL-3233/bring-back-sms-reminders-for-teams)</sub>

<sub>[CAL-3430](https://linear.app/calcom/issue/CAL-3430/[cal-3233]-bring-back-sms-reminders-for-teams)</sub>

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Before SMS/Whatsapp to attendee was only available for Orgs, now it will be available for all teams. It adds rate limit to sending SMS + an Admin view (localhost:3000/settings/admin/lockedSMS) to lock users and teams to send SMS.

Fixes #13897

### CODE REVIEW

File: packages/features/ee/workflows/lib/reminders/providers/twilioProvider.ts
Comment (joeauyeung) [inline packages/features/ee/workflows/lib/reminders/providers/twilioProvider.ts:178]: NIT: Can we actually use an `OR` condition in the prisma query against the `userId` to see if the user is locked.

**Review by joeauyeung — DISMISSED** [review]
Everything looks good! Nothing has really changed code wise in terms of sending SMS/Whatsapp messages it was just disabling the feature. I have a few nits but nothing blocking.

File: packages/features/ee/workflows/lib/reminders/reminderScheduler.ts
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/reminders/reminderScheduler.ts]: Also team workflows have a `userId` (the user that created the workflow). We need to check for `teamId` here

File: packages/lib/checkRateLimitAndThrowError.ts
Comment (CarinaWolli) [inline packages/lib/checkRateLimitAndThrowError.ts:61]: @joeauyeung  don't lock user's that belong to an org

File: packages/lib/checkRateLimitAndThrowError.ts
Comment (CarinaWolli) [inline packages/lib/checkRateLimitAndThrowError.ts:71]: @joeauyeung  don't lock orgs or org teams

File: packages/features/ee/workflows/lib/reminders/reminderScheduler.ts
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/reminders/reminderScheduler.ts:51]: Also team workflows have a `userId` (the user that created the workflow). We need to check for `teamId` here

File: apps/web/pages/settings/admin/lockedSMS/UsersTable.tsx
Comment (Udit-takkar) [inline apps/web/pages/settings/admin/lockedSMS/UsersTable.tsx:104]: we should use `getUserAvatarUrl` here

**Review by Udit-takkar — COMMENTED** [review]
can you also block orgs teams?

Getting this error

Screenshot 2024-04-22 at 8 20 24 PM

File: apps/web/pages/settings/admin/lockedSMS/UsersTable.tsx
Comment (Udit-takkar) [inline apps/web/pages/settings/admin/lockedSMS/UsersTable.tsx:128]: here we should probably use  const profileImageSrc = getPlaceholderAvatar(team.logoUrl || team.parent?.logoUrl, team.name);

**Comment by CarinaWolli** [comment]
> can you also block orgs teams?

orgs can't be blocked. also not their users and teams

File: apps/web/pages/settings/admin/lockedSMS/UsersTable.tsx
Comment (Udit-takkar) [inline apps/web/pages/settings/admin/lockedSMS/UsersTable.tsx:29]: Not urgent: use infinite query here. we can probably use it later

**Review by Udit-takkar — COMMENTED** [review]
LGTM. works fine for me. will wait for @joeauyeung re review before merging

**Review by joeauyeung — APPROVED** [review]
No big changes since my last approval. Approving again.

**Comment by keithwillcode** [comment]
@CarinaWolli @joeauyeung @Udit-takkar why was this PR merged with 0 tests added to verify SMS locking logic?

**Pull request merged**
